pkg/snmp/traps/listener_test.go

```diff
@@ -17,19 +17,29 @@ import (
 )
 
 var serverPort = getFreePort()
+var initialTrapsPacketsAuthErrors int64
 
 const defaultTimeout = 1 * time.Second
 
-func TestListenV1GenericTrap(t *testing.T) {
+func listenerTestSetup(t *testing.T, config Config) (*mocksender.MockSender, *TrapListener) {
 	mockSender := mocksender.NewMockSender("snmp-traps-telemetry")
 	mockSender.SetupAcceptAll()
+	packetOutChan := make(PacketsChannel, packetsChanSize)
 
-	config := Config{Port: serverPort, CommunityStrings: []string{"public"}, Namespace: "totoro"}
 	Configure(t, config)
 
-	packetOutChan := make(PacketsChannel, packetsChanSize)
 	trapListener, err := startSNMPTrapListener(config, mockSender, packetOutChan)
 	require.NoError(t, err)
+
+	// trapsPacketsAuthErrors is global so its value carries over from test to test.  Capture its initial value to determine if it changes during an individual test run.
+	initialTrapsPacketsAuthErrors = trapsPacketsAuthErrors.Value()
```

Comment:
This was needed at the beginning of every test, and I also combined other common setup code into listenerTestSetup().